prime-router/src/main/kotlin/Report.kt

```diff
@@ -699,10 +718,22 @@ class Report {
             createdDateTime: OffsetDateTime,
             nameFormat: NameFormat = NameFormat.STANDARD,
             receivingOrganization: String? = null,
-            sendingFacility: String = "cdcprime"
+            sendingFacility: String = "cdcprime",
+            processingModeCode: String = "T",
+            translationConfig: TranslatorConfiguration? = null,
         ): String {
+            fun mapProcessingModeCode(processingModeCode: String = "T"): String {
+                return when (processingModeCode.toLowerCase()) {
+                    "p" -> "production"
+                    "d" -> "development"
+                    else -> "testing"
+                }
+            }
             val formatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmss")
```

Comment:
Moved the mapping of the processing mode code into a separate internal function.